### PR TITLE
Changed build to run all npm builds before backend builds.

### PR DESCRIPTION
**Pull request by valadas**
This makes `yarn install` and `yarn build` before any project builds. It still uses lerna as before.

- I confirmed that a failed install makes the build fail.
- I confirmed that a failed npm build makes the build fail.
- I confirmed that it fixed the resource manager scripts missing from the final build.

The drawback of this approach is that one can no longer build only the persona bar in release mode without using cake. But for dev/debug, the current documented way with `yarn start --scope ...` is still valid.

### CODE REVIEW

**Review by david-poindexter — APPROVED** [review]
Thank you @valadas - this looks like the lessor of evils here.

File: DNN_Platform.sln [hunk 1]
```diff
@@ -500,6 +500,9 @@ EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "DotNetNuke.Providers.Caching.SimpleWebFarmCachingProvider", "DNN Platform\Providers\CachingProviders\DotNetNuke.Providers.Caching.SimpleWebFarmCachingProvider\DotNetNuke.Providers.Caching.SimpleWebFarmCachingProvider.csproj", "{2C25580C-A971-4F0B-9F70-436A35C2473E}"
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Dnn.Modules.ResourceManager", "DNN Platform\Modules\ResourceManager\Dnn.Modules.ResourceManager.csproj", "{7D61A32C-0F21-453F-A981-BD8E5A3A5304}"
+	ProjectSection(ProjectDependencies) = postProject
+		{9CCA271F-CFAA-42A3-B577-7D5CBB38C646} = {9CCA271F-CFAA-42A3-B577-7D5CBB38C646}
+	EndProjectSection
```
Comment (bdukes) [inline DNN_Platform.sln:505]: Do we know what this change is?

File: Build/BuildScripts/AEModule.build [hunk 2]
```diff
@@ -1,14 +1,12 @@
 ﻿<Project ToolsVersion="4.0" DefaultTargets="Build" 
   xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="..\..\packages\Yarn.MSBuild.1.22.17\build\Yarn.MSBuild.props" Condition="Exists('..\..\packages\Yarn.MSBuild.1.22.17\build\Yarn.MSBuild.props')" />
-  <Import Project="..\..\packages\Yarn.MSBuild.1.22.17\build\Yarn.MSBuild.targets" Condition="Exists('..\..\packages\Yarn.MSBuild.1.22.17\build\Yarn.MSBuild.targets')" />
```
Comment (bdukes) [inline Build/BuildScripts/AEModule.build:4]: Should we remove the `Yarn.MSBuild` package entirely?

Comment (valadas) [reply Build/BuildScripts/AEModule.build:4]: Well, if we go with this solution then it is unused...

File: DNN_Platform.sln (hunk 1 above)
Comment (valadas) [reply DNN_Platform.sln:505]: I have no clue, it was done by Visual Studio...

**Review by donker — APPROVED** [review]
I quite like moving the npm build from being hidden in some build file to the main scripts

**Pull request merged**